### PR TITLE
22 modif evento

### PR DESCRIPTION
**Pull request by martareluy**
# Description
**No pasa el test, pero no se hacerlo, se aceptan sugerencias :)** Todo lo demás hecho, tambien el merge con la última version de main para que no de problemas :p 
## Context

(Optional)

Current situation of the system, explanation why this PR is needed.

## Goal

Explanation about what we want to achieve with this PR.

## Support documents

- Taiga Ticket: [ticket](link)

- UI/UX design: [link](https://paack.co)

## Design considerations

(Optional)

Section dedicated to explain what design considerations we had while coding this PR.

# Testing

Section dedicated to explain how and what to test in this PR.

# Configuration

## New environment

(Optional)

Environment variables added/changed that needs to be added manually by the other developers

## New dependencies

(Optional)

New packages/libraries added / updated

# Attachments

(Optional)

- <details closed>
    
  </details>

Section dedicated to add any screenshots or other documents we find relevant to help understand or analyze this PR.

# Dependencies or blockers

(Optional: List of dependencies or blockers that make this PR not yet ready to be merged with the main branch)
## Subtasks

(List of subtask PRs needed for this feature)

- [ ] #XX
...

### CODE REVIEW

File: src/infrastructure/dtos/edit-event.dto.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    IsDate,
+    Max,
+    IsNumber,
+    Min,
+    IsOptional
+  } from 'class-validator';
+  import { validate } from 'class-validator';
+  import type { NextFunction, Request, Response } from 'express';
+
+export class EditParamDTO {
```
Comment (quimvi) [inline src/infrastructure/dtos/edit-event.dto.ts:13]: Millor dir-li editEventDTO

File: src/application/controllers/event-controller/event-controller.ts [hunk 2]
```diff
@@ -48,6 +48,32 @@ export class EventController {
     }
   }
 
+  public static async editEvent(req: Request, res: Response): Promise<void> {
```
Comment (quimvi) [inline src/application/controllers/event-controller/event-controller.ts:51]: Crec que aqui faria falta veure que passa si no troba l'event que es vol modificar

File: src/infrastructure/api/event-router/event-router.test.ts [hunk 3]
```diff
@@ -65,4 +65,67 @@ describe('User Routes', function () {
...
+
+  
+      // Verifica que la respuesta de la solicitud sea la esperada y que el evento se haya modificado correctamente
+      expect(res.statusCode).toBe(200);
+      expect(res.body.message).toBe('Evento editado correctamente');
+  
+      //const updatedEvent = await Event.to(12348173049);
+  
+      //expect(res.supdatedEvent.denominacio).toBe('new-test-event');
+      //expect(res.updatedEvent.descripcio).toBe('new-test-description');
```
Comment (quimvi) [inline src/infrastructure/api/event-router/event-router.test.ts:102]: Borra els comments

File: src/infrastructure/api/event-router/event-router.test.ts [hunk 4]
```diff
@@ -65,4 +65,67 @@ describe('User Routes', function () {
...
+        url: 'https://test-url.com',
+      });
+  
+      const res = await request(app).post('/events/edit').send({
+        codi: 123456789,
+        denominacio: '',
+      });
+  
+      expect(res.statusCode).toBe(400);
+      expect(res.body.success).toBe(false);
```
Comment (quimvi) [inline src/infrastructure/api/event-router/event-router.test.ts:125]: El codi et peta perk no te un camp de success al body si el payload es incorrecte.
Hauries de comprobar que el camp de errors te coses. per exemple:
```suggestion
      expect(res.body.errors).toBeTruthy()
```

File: src/infrastructure/dtos/edit-event.dto.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    @IsNotEmpty()
+    @IsNumber()
+    @Min(11111111111)
+    @Max(99999999999)
+    codi: number;
+  
+    @IsString()
+    @IsOptional()
+    denominacio: string | null;
```
Comment (quimvi) [inline src/infrastructure/dtos/edit-event.dto.ts:22]: Si es optional no cal posar `string | null`. Si poses que es optional ja dius que pot ser  null

File: src/application/controllers/event-controller/event-controller.ts (hunk 2 above)
Comment (quimvi) [reply src/application/controllers/event-controller/event-controller.ts:51]: Si fas aixo tambe posa un test mes per intentar modificar un event que no existeix

**Review by quimvi — CHANGES_REQUESTED** [review]

**Review by quimvinals — APPROVED** [review]

**Review by quimvi — APPROVED** [review]

**Pull request merged**